### PR TITLE
[stale] Feature/message schema

### PR DESCRIPTION
**Pull request by fletcher-greg**
I've created a message schema to review.  

This model seems to make the most sense to me.  

Every message is stored with a sender and receiver.   This way we can retrieve each message and serve the data to both users that need to see it

### CODE REVIEW

File: src/resources/components/message/Message.model.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        },
+        sender: {
+          type: Schema.Types.ObjectId,
+          ref: 'User',
+          required: true
+        }
+      }
+    ],
+
+    read: { type: Date }
```
Comment (SebastianLF) [inline src/resources/components/message/Message.model.js:26]: What does this 'read' property is used for?

File: src/utils/auth.js [hunk 2]
```diff
@@ -12,9 +13,14 @@ function newToken(id) {
 
 async function signup(req, res) {
   try {
+    //  save user
     const userSaved = await User.create(req.body)
+    //  create JWT Token
     const token = newToken(userSaved._id)
-    return res.status(200).json({ token })
+    //  create trainer
+    const trainerSave = await Trainer.create(req.body.trainer)
```
Comment (SebastianLF) [inline src/utils/auth.js:21]: By doing this, a trainer is created everytime /signup/ route is hitted but it's not working for the case when we want to simply create a client account (that is to say not creating a trainer account).
```
if(singupForTraineraccount) {
  const trainerSave = await Trainer.create(req.body.trainer)
}
```
Also, The following writes directly to MongoDB and should usually be avoided unless certain cases like writing tests or other cases when you deliberately want to write in invalid data.
`Trainer.create(req.body.trainer)`
check this: https://twm.me/correct-way-to-use-mongoose/

File: src/resources/components/trainer/trainer.model.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
+  gender: String,
+  location: { type: String, required: true },
+  avatar: String,
+  presentation: String,
+  specialities: [String],
+  reviews: [],
+  conversations: [],
+  contacts: []
+})
```
Comment (SebastianLF) [inline src/resources/components/trainer/trainer.model.js:13]: What do you think about the way we have modelised Trainer? @lunify @Moniet

File: src/resources/components/message/Message.model.js (hunk 1 above)
Comment (fletcher-greg) [reply src/resources/components/message/Message.model.js:26]: That 'read' property is to show when the message was read by the receiver.

File: src/utils/auth.js (hunk 2 above)
Comment (fletcher-greg) [reply src/utils/auth.js:21]: Yes, I agree we don't want to create a trainer every time a user signs up.  I've put that there just for testing purposes.  

I'll update the code to reflect your comments.  😊

File: src/resources/components/message/Message.model.js (hunk 1 above)
Comment (SebastianLF) [reply src/resources/components/message/Message.model.js:26]: That's a great idea! so a 'sender' property is needed inside 'message:' object. This way we can identify who is the sender and who is the receiver.
```
message: {
  text: { type: String, required: true },
  sender: { type: String, required: true }
}
```

**Comment by fletcher-greg** [comment]
Closed.  Using discussion schema now

**Comment by fletcher-greg** [comment]
Closing.  Using discussion schema for now

**Pull request closed without merging**